gateway/server.go

```diff
@@ -665,6 +642,55 @@ func (gw *Gateway) loadControlAPIEndpoints(muxer *mux.Router) {
 
 	r.HandleFunc("/schema", gw.schemaHandler).Methods(http.MethodGet)
 
+	gw.controlAPIRouter = r
+	return r
+}
+
+// loadControlAPIEndpoints loads the endpoints used for controlling the Gateway.
+func (gw *Gateway) loadControlAPIEndpoints(muxer *mux.Router) {
+	hostname := gw.GetConfig().HostName
+	if gw.GetConfig().ControlAPIHostname != "" {
+		hostname = gw.GetConfig().ControlAPIHostname
+	}
+
+	if muxer == nil {
+		cp := gw.GetConfig().ControlAPIPort
```

Comment:
I don't like the indentation here, it doesn't look like a good API with several fallbacks. A smaller function for this block would clear up the logic.

Something like:
`func (gw *Gateway) getProxyMux() (*mux.Router, error)`